code/controllers/subsystems/supply.dm

```diff
@@ -264,16 +285,61 @@ SUBSYSTEM_DEF(supply)
 	var/reason = null
 	var/orderedrank = null //used for supply console printing
 
-/datum/controller/subsystem/supply/proc/find_item_value(var/obj/object) //here we get the value of the items being traded
+/datum/controller/subsystem/supply/proc/make_trade(var/obj/object, var/count = 1)
+	. = find_item_value(object, count)
+	if(.)
+		sold_items[object.type] += count
+	return .
+
+/datum/controller/subsystem/supply/proc/find_item_value(var/obj/object, var/count = 1, var/use_reinf_material = FALSE) //here we get the value of the items being traded
 	if(!object)
 		return 0
 
-	//this uses the default SS13 item_worth procs so its a good fallback
-	. = get_value(object)
+	var/sell_modifier = 1
 
-	var/datum/trade_item/T
+	if(GLOB.using_map.using_new_cargo)
```

Comment:
If you make the above changes, payout() will only be called if the map is using_new_cargo, which is how it should be, so you could remove the check here, or put it at the top of the proc instead

Reply:
Re-wrote this part to clean it up a little. Check has been moved to within the type check for stacks. It's still required there as stacks are sold in both the old and new system but at least this way it's not constantly being checked when it's unnecessary.

Reply:
To clarify, material stacks are sold slightly differently under the old system, and the actual "points" are added separately from trading points on line 210. I haven't looked through your changes fully yet, but doing it this way might end up with materials being sold twice on Glloydstation.

Reply:
I'm rather confused. Comparing the old file to the new changes, nothing should be any different. In the old file material stacks were tallied https://github.com/UristMcStation/UristMcStation/blob/a8182098b0e1dad3c201e47b16b996e1c5a67689/code/controllers/subsystems/supply.dm#L142 & https://github.com/UristMcStation/UristMcStation/blob/a8182098b0e1dad3c201e47b16b996e1c5a67689/code/controllers/subsystems/supply.dm#L173

In fact, looking at this code, materials _were_ actually being sold twice if you were using the new cargo system due to a lack of a `continue` or `else` on L150 or L164 respectively.

Both of these lines tallied the value using slightly different methods --one using cargo inflation and the other the material multiplier, which has been carried over in the new code on L303-- and added to the same list, `material_count` which was then processed at the end (L197 on the old, L210 new). The method of calculating the value has simply been moved outside of the sell loop and into `find_item_value()` for ease of future changes. Again, on the old file both new cargo and old added the materials to the same list which was then processed in the same way without a further new cargo check. I'm really struggling to see how this new code functions any differently unless I'm being incredibly dumb?